### PR TITLE
style: use ruff instead of prospector

### PR DESCRIPTION
**Pull request by DaniBodor**

### CODE REVIEW

**Comment by wbaccinelli** [comment]
To be checked the PathLib issie

**Comment by DaniBodor** [comment]
> To be checked the PathLib issie

Could you please test it again on the last commit, and if that doesn't work on the commit before ([1dbf35f](https://github.com/EIT-ALIVE/eit_dash/pull/28/commits/1dbf35f1550e345c5b78df6eed62b98a210984f2)).
If neither of those work, then I will just move back to using `os` instead of `pathlib`.

Semi-related to this: is the `get_parent_directory` function doing anything at this point? It is not referenced in the rest of the code, and it seems I can remove it and the dashboard still works.

**Comment by wbaccinelli** [comment]
> > To be checked the PathLib issie
> 
> Could you please test it again on the last commit, and if that doesn't work on the commit before ([1dbf35f](https://github.com/EIT-ALIVE/eit_dash/pull/28/commits/1dbf35f1550e345c5b78df6eed62b98a210984f2)). If neither of those work, then I will just move back to using `os` instead of `pathlib`.
> 
I tried on both the commits, but I get the same error. The problem was that the with Path the output was not a string but a path object an this is not accepted by the some of the dash core components. I am pushing the changes.

> Semi-related to this: is the `get_parent_directory` function doing anything at this point? It is not referenced in the rest of the code, and it seems I can remove it and the dashboard still works.

Yes, it is used when you click "parent directory" in the popup. Unfortunately, with Dash it's not easy to retrieve immediately if a callback function is actually used, because it is not directly referenced in the code, but it's triggered when the input in the decorator is changed.

**Comment by DaniBodor** [comment]
> I tried on both the commits, but I get the same error. The problem was that the with Path the output was not a string but a path object an this is not accepted by the some of the dash core components. I am pushing the changes.

Thanks! I squashed the three commits regarding pathlib into a single commit as they together fixed a single thing.

**Comment by DaniBodor** [comment]
@wbaccinelli , I still need you to approve the PR before I can merge :)

**Review by wbaccinelli — APPROVED** [review]
Tested and it works :)

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,9 +1,36 @@
...
+  },
+  "autoDocstring.docstringFormat": "google",
+
+  // Notebooks
+  "notebook.lineNumbers": "on",
+  "notebook.formatOnSave.enabled": true,
+  "notebook.codeActionsOnSave": {
+    "notebook.source.fixAll": "explicit"
+  },
+  "notebook.diff.ignoreMetadata": true,
```
[the remark this answers is not in the document]
Comment (DaniBodor) [reply .vscode/settings.json:19]: agreed!

**Pull request merged**